File: cyclic_rotation/opussf/cyc.lua
#!/usr/bin/env lua

function solution( tableIn, shift )
	for lcv=1, shift do
		table.insert( tableIn, 1, table.remove( tableIn ) )
	end
	return tableIn
end

assert( solution( {3, 8, 9, 7, 6}, 1 )[1] == 6 )
assert( solution( {3, 8, 9, 7, 6}, 3 )[1] == 9 )
assert( solution( {3, 8, 9, 7, 6}, 0 )[1] == 3 )

File: cyclic_rotation/opussf/cyc.py
#!/usr/bin/env python

def solution( listIn, shiftVal ):
	outList = listIn[-shiftVal:] + listIn[:-shiftVal]
	return outList

if __name__=="__main__":
	import unittest

	class testCyclicShift( unittest.TestCase ):
		def setUp( self ):
			pass
		def test01( self ):
			self.assertEquals( solution( [ 3, 8, 9, 7, 6 ], 1 ), [ 6, 3, 8, 9, 7 ] )
		def test02( self ):
			self.assertEquals( solution( [ 3, 8, 9, 7, 6 ], 3 ), [ 9, 7, 6, 3, 8 ] )
		def test03( self ):
			self.assertEquals( solution( [ 3, 8, 9, 7, 6 ], 0 ), [ 3, 8, 9, 7, 6 ] )
		def test04( self ):  # Bonus!  it goes 'backwards' too
			self.assertEquals( solution( [ 3, 8, 9, 7, 6 ], -2 ), [ 9, 7, 6, 3, 8 ] )

	unittest.main()
